tests/zfs-tests/tests/functional/cli_root/zpool_events/zpool_events_errors.ksh

```diff
@@ -0,0 +1,144 @@
+#!/bin/ksh -p
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2018 by Lawrence Livermore National Security, LLC.
+#
+
+# DESCRIPTION:
+#	Verify the number of IO and checksum events match the error counters
+#	in zpool status.
+#
+# STRATEGY:
+#	1. Create a raidz or mirror pool
+#	2. Inject read/write IO errors or checksum errors
+#	3. Verify the number of errors in zpool status match the corresponding
+#	   number of error events.
+#	4. Repeat for all combinations of raidz/mirror and io/checksum errors.
+#
+
+. $STF_SUITE/include/libtest.shlib
+
+verify_runnable "both"
+
+MOUNTDIR=$TEST_BASE_DIR/mount
+VDEV1=$TEST_BASE_DIR/file1
+VDEV2=$TEST_BASE_DIR/file2
+VDEV3=$TEST_BASE_DIR/file3
+POOL=error_pool
+POOLTYPE=raidz
+FILESIZE=$((5 * 1024 * 1024))
+OLD_CHECKSUMS=$(get_tunable zfs_checksum_events_per_second)
+OLD_LEN_MAX=$(get_tunable zfs_zevent_len_max)
+
+function cleanup
+{
+	log_must set_tunable64 zfs_checksum_events_per_second $OLD_CHECKSUMS
+	log_must set_tunable64 zfs_zevent_len_max $OLD_LEN_MAX
+
+	log_must zinject -c all
+	log_must zpool events -c
+	if poolexists $POOL ; then
+		log_must destroy_pool $POOL
+	fi
+	log_must rm -f $VDEV1 $VDEV2 $VDEV3
+}
+
+log_assert "Check that the number of zpool errors match the number of events"
+
+log_onexit cleanup
+
+set_tunable64 zfs_checksum_events_per_second 1000
+set_tunable64 zfs_zevent_len_max 1000
```

Comment:
A comment here, or above, explaining why these need to be changed would be good.